Mage.Sets/src/mage/cards/h/HinataDawnCrowned.java

```diff
@@ -0,0 +1,122 @@
+
+        // Trample
+        this.addAbility(TrampleAbility.getInstance());
+
+        // Spells you cast cost {1} less to cast for each target.
+        this.addAbility(new SimpleStaticAbility(new HinataDawnCrownedOwnEffect()));
+
+        // Spells your opponents cast cost {1} more to cast for each target
+        this.addAbility(new SimpleStaticAbility(new HinataDawnCrownedOpponentsEffect()));
+    }
+
+    private HinataDawnCrowned(final HinataDawnCrowned card) {
+        super(card);
+    }
+
+    @Override
+    public HinataDawnCrowned copy() {
+        return new HinataDawnCrowned(this);
+    }
+}
+
+class HinataDawnCrownedOwnEffect extends CostModificationEffectImpl {
+
+    HinataDawnCrownedOwnEffect() {
+        super(Duration.WhileOnBattlefield, Outcome.Neutral, CostModificationType.REDUCE_COST);
+        staticText = "Spells you cast cost {1} less to cast for each target";
+    }
+
+    private HinataDawnCrownedOwnEffect(HinataDawnCrownedOwnEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public HinataDawnCrownedOwnEffect copy() {
+        return new HinataDawnCrownedOwnEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source, Ability abilityToModify) {
+        CardUtil.reduceCost(abilityToModify, HinataDawnCrownedEffectUtility.getTargetCount(game, abilityToModify));
+        return true;
+    }
+
+    @Override
+    public boolean applies(Ability abilityToModify, Ability source, Game game) {
+        return abilityToModify instanceof SpellAbility
+                && abilityToModify.isControlledBy(source.getControllerId());
+    }
+}
+
+class HinataDawnCrownedOpponentsEffect extends CostModificationEffectImpl {
+
+    HinataDawnCrownedOpponentsEffect() {
+        super(Duration.WhileOnBattlefield, Outcome.Neutral, CostModificationType.INCREASE_COST);
+        staticText = "Spells your opponents cast cost {1} more to cast for each target";
+    }
+
+    private HinataDawnCrownedOpponentsEffect(HinataDawnCrownedOpponentsEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public HinataDawnCrownedOpponentsEffect copy() {
+        return new HinataDawnCrownedOpponentsEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source, Ability abilityToModify) {
+        CardUtil.increaseCost(abilityToModify, HinataDawnCrownedEffectUtility.getTargetCount(game, abilityToModify));
+        return true;
+    }
+
+    @Override
+    public boolean applies(Ability abilityToModify, Ability source, Game game) {
+        return abilityToModify instanceof SpellAbility
+                && !abilityToModify.isControlledBy(source.getControllerId());
+    }
+}
+
+final class HinataDawnCrownedEffectUtility
+{
+    public static int getTargetCount(Game game, Ability abilityToModify)
+    {
+        return (int)(game.inCheckPlayableState() ?
+                CardUtil.getAllPossibleTargets(abilityToModify, game).stream().count():
```

Comment:
I think on too many possible targets it will never works. Example:
- you can choose max 1 target;
- you have 10 possible targets;
- current code increase cost by 10 instead 1

Each target can have max limit. So playable state can check Math.min(allPossible, maxPossibleTargetSize) or like that. So playable check will see min possible cost increase, not maximum.

Reply:
I see what you are saying. Do you happen to know how I would go about getting the maxPossibleTargetSize?

Reply:
Created a new [PR](https://github.com/magefree/mage/pull/8657) with this fix